[BUG] premerge fails dues to Spark 3.3.0 HadoopFsRelation after SPARK-37289

**Describe the bug**
```
 def partitionSchemaOption: Option[StructType] is removed
```
**Steps/Code to reproduce bug**
```shell
$ ./build/buildall --profile=330

[ERROR] [Error] /home/gshegalov/gits/NVIDIA/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala:371: value partitionSchemaOption is not a member of org.apache.spark.sql.execution.datasources.HadoopFsRelation
[ERROR] [Error] /home/gshegalov/gits/NVIDIA/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala:348: value partitionSchemaOption is not a member of org.apache.spark.sql.execution.datasources.HadoopFsRelation
[ERROR] two errors found
```

**Expected behavior**
working build

**Environment details (please complete the following information)**
local and CI

--- a/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala
+++ b/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala
@@ -345,7 +345,7 @@ case class GpuFileSourceScanExec(
       driverMetrics("staticFilesNum") = filesNum
       driverMetrics("staticFilesSize") = filesSize
     }
-    if (relation.partitionSchemaOption.isDefined) {
+    if (relation.partitionSchema.nonEmpty) {
       driverMetrics("numPartitions") = partitions.length
     }
   }
@@ -368,7 +368,7 @@ case class GpuFileSourceScanExec(
       None
     }
   } ++ {
-    if (relation.partitionSchemaOption.isDefined) {
+    if (relation.partitionSchema.nonEmpty) {
       Map(
         NUM_PARTITIONS -> createMetric(ESSENTIAL_LEVEL, DESCRIPTION_NUM_PARTITIONS),
         "pruningTime" -> createTimingMetric(ESSENTIAL_LEVEL, "dynamic partition pruning time"))
@@ -494,7 +494,7 @@ case class GpuFileSourceScanExec(
       fsRelation: HadoopFsRelation,
       readFile: Option[(PartitionedFile) => Iterator[InternalRow]],
       partitions: Seq[FilePartition]): RDD[InternalRow] = {
-    
+
     if (isPerFileReadEnabled) {
       logInfo("Using the original per file parquet reader")
       ShimLoader.getSparkShims.getFileScanRDD(fsRelation.sparkSession, readFile.get, partitions)

